### PR TITLE
CDAP-8559: Improve metadata error message

### PR DESCRIPTION
**Pull request by rohitsinha54**
Issue: https://issues.cask.co/browse/CDAP-8559

Build: https://builds.cask.co/browse/CDAP-RUT1624-1

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/metadata/DefaultMetadataAdmin.java [hunk 1]
```diff
@@ -203,7 +203,10 @@ private void validateTagReservedKey(MetadataEntity metadataEntity, String key)
   private void validateKeyAndTagsFormat(MetadataEntity metadataEntity, String keyword)
     throws InvalidMetadataException {
     if (!KEY_AND_TAG_MATCHER.matchesAllOf(keyword)) {
-      throw new InvalidMetadataException(metadataEntity, "Illegal format for the value : " + keyword);
+      throw new InvalidMetadataException(metadataEntity, String.format("Illegal format for the value '%s'. Must only " +
+                                                                         "consists of characters[a-z or A-Z], " +
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/metadata/DefaultMetadataAdmin.java]: @rohitsinha54 I think this is matching the key, not the value, so the error message is very misleading. Should we instead say ``Illegal metadata property '%s'. Properties can only contain alphanumeric characters (a-z, A-Z, 0-9), underscores and hyphens.``

Also, If you look at the jira, there is a long text preceding this actual error message. Can that be fixed? Particularly, I think the ``failed due to error`` portion should be removed.

Does that come from the backend or the UI? @elfenheart FYI

Comment (rohitsinha54) [reply cdap-app-fabric/src/main/java/co/cask/cdap/metadata/DefaultMetadataAdmin.java]: Value here is used in generic sense I believe. This message is for both 'value' in key-value metadata properties and and tag.

I don't think we should say property in message as in metadata it has specific meaning and this message is common for tag too. 

We can just say Illegal format for '<user-data>'

File: cdap-app-fabric/src/main/java/co/cask/cdap/metadata/DefaultMetadataAdmin.java [hunk 2]
```diff
@@ -216,10 +216,10 @@ private void validateKeyAndTagsFormat(MetadataEntity metadataEntity, String keyw
   private void validateValueFormat(MetadataEntity metadataEntity, String keyword)
     throws InvalidMetadataException {
     if (!VALUE_MATCHER.matchesAllOf(keyword)) {
-      throw new InvalidMetadataException(metadataEntity, String.format("Illegal format for the value '%s'. Must only " +
-                                                                         "consists of characters[a-z or A-Z], " +
-                                                                         "digits[0-9], underscore(_), hyphen(-) or " +
-                                                                         "whitespace.", keyword));
+      throw new InvalidMetadataException(metadataEntity, String.format("Illegal format for the value '%s'. Should " +
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/metadata/DefaultMetadataAdmin.java:219]: should we remove value from here too? can this ever be a tag?

**Review by bdmogal — APPROVED** [review]
one question, otherwise lgtm

File: cdap-app-fabric/src/main/java/co/cask/cdap/metadata/DefaultMetadataAdmin.java (hunk 2 above)
Comment (rohitsinha54) [reply cdap-app-fabric/src/main/java/co/cask/cdap/metadata/DefaultMetadataAdmin.java:219]: This is the value of the key-value property. This cannot be a tag. Tag needs to follow the pattern of KEY_AND_TAG_MATCHER

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by rohitsinha54** [comment]
Waiting on build: https://builds.cask.co/browse/CDAP-RUT1624-4

**Pull request merged**